File: frontend/src/lib/roleStorage.ts
const ROLE_KEY = 'employabilityos_role'
const DEFAULT_ROLE = 'Frontend Developer'

const ROLE_MAP: Record<string, string> = {
  'Frontend Engineer': 'Frontend Developer',
  'Backend Engineer': 'Backend Developer',
  'Data Analyst': 'Data Analyst',
  'AI/ML Engineer': 'AI/ML Engineer',
  'Mobile Developer': 'Mobile Developer',
  'Product Designer': 'Frontend Developer',
  'frontend': 'Frontend Developer',
  'backend': 'Backend Developer',
  'data-analyst': 'Data Analyst',
  'ai-ml': 'AI/ML Engineer',
  'mobile': 'Mobile Developer',
  'designer': 'Frontend Developer',
}

export function getStoredRole(): string {
  try {
    return localStorage.getItem(ROLE_KEY) || DEFAULT_ROLE
  } catch {
    return DEFAULT_ROLE
  }
}

export function setStoredRole(role: string) {
  const mapped = ROLE_MAP[role] || role
  localStorage.setItem(ROLE_KEY, mapped)
}

File: android-app/app/src/main/java/com/skilljobgap/employability/data/RoleStorage.kt
package com.skilljobgap.employability.data

import android.content.Context
import android.content.SharedPreferences

/**
 * Mirrors web roleStorage: getStoredRole / setStoredRole.
 * Role is read from users.role (ProfileRepository) and cached here
 * so all role-dependent screens use the same value.
 */
object RoleStorage {

    private const val PREFS_NAME = "employabilityos_prefs"
    private const val KEY_ROLE = "employabilityos_role"
    private const val DEFAULT_ROLE = "Frontend Developer"

    private fun prefs(context: Context): SharedPreferences =
        context.applicationContext.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    fun getStoredRole(context: Context): String =
        prefs(context).getString(KEY_ROLE, DEFAULT_ROLE) ?: DEFAULT_ROLE

    fun setStoredRole(context: Context, role: String) {
        val dbRole = ROLE_MAP[role] ?: role
        prefs(context).edit().putString(KEY_ROLE, dbRole).apply()
    }

    /** Map UI/alias role labels to Supabase users.role value. */
    private val ROLE_MAP = mapOf(
        "Frontend" to "Frontend Developer",
        "Backend" to "Backend Developer",
        "Data Analyst" to "Data Analyst",
        "AI/ML" to "AI/ML Engineer",
        "Mobile" to "Mobile Developer",
        "frontend" to "Frontend Developer",
        "backend" to "Backend Developer",
        "data-analyst" to "Data Analyst",
        "ai-ml" to "AI/ML Engineer",
        "mobile" to "Mobile Developer"
    )
}
